src/main/java/dev/j3fftw/litexpansion/machine/Generator.java

```diff
@@ -0,0 +1,66 @@
+package dev.j3fftw.litexpansion.machine;
+
+import dev.j3fftw.litexpansion.Items;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AGenerator;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public class Generator extends AGenerator {
+
+    public Generator() {
+        super(Items.LITEXPANSION, Items.GENERATOR, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            null, Items.RE_BATTERY, null,
+            null, Items.MACHINE_BLOCK, null,
+            null, new ItemStack(Material.FURNACE), null
+        });
+    }
+
+    @Nonnull
+    @Override
+    public String getInventoryTitle() {
+        return "&7Generator";
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.FIRE);
```

Comment:
Did you test this? Because FIRE has been illegal for ItemStacks for some time now.